### ISSUE TITLE
Dalvik-appeasing names

### ISSUE THREAD

**Issue by tombentley** [ceylon/ceylon-compiler#2152]
As mentioned [here](https://groups.google.com/forum/?utm_medium=email&utm_source=footer#!msg/ceylon-users/MRPDP8LfHX0/7tPo7_br4AMJ), it seems dalvik has a problem with some fields names we generate. Although they seem to be acceptable according to the [JVM Spec §4.2.2]9http://docs.oracle.com/javase/specs/jvms/se7/html/jvms-4.html#jvms-4.2.2) we should try to make this work on dalvik.

**Comment by lucaswerkmeister** [comment]
This will break binary compatibility, won’t it?

**Comment by tombentley** [comment]
Yes, but it would only affect people who were reflectively inspecting the annotations of `Import`s, which I can't imagine is very many people. I think the work @FroMage has been doing on annotations has already broken BC in a much more major way.

**Comment by lucaswerkmeister** [comment]
> I think the work @FroMage has been doing on annotations has already broken BC in a much more major way.

Surely not for 1.3 ;)
